### ISSUE TITLE
Count operator doesn't accept variables

### ISSUE THREAD

**Issue by theseion** [coreruleset/secrules_parsing#70]
The count operator does not accept variables. Plugging `Variable` in there leads to infinite recursion, and using `VariableName` breaks other tests.

Example test:

```python
def test_variable_allowed_as_count_arg() -> None:
    """Test that variables are allowed as arguments to `&`"""
    rule_text = """
    SecRule &REQUEST_BODY "@eq 1" \
     "id:941181,\
     phase:2,\
     block,\
     capture,\
     t:none"
     """

    parsed_rule = parser.process_from_str(rule_text)
    print(parsed_rule)
    assert type(parsed_rule) is not dict
```

Line at issue: https://github.com/coreruleset/secrules_parsing/blob/88ef01a920767f5eede1f4344c345db7206491e6/src/secrules_parsing/model/secrules.tx#L72.

**Comment by airween** [comment]
This quick (and ugly) hack solves the issue:

```diff
diff --git a/src/secrules_parsing/model/secrules.tx b/src/secrules_parsing/model/secrules.tx
index 56ef6ee..ebcd1fc 100644
--- a/src/secrules_parsing/model/secrules.tx
+++ b/src/secrules_parsing/model/secrules.tx
@@ -50,7 +50,7 @@ There is no check against collections existance, or typying between variables an
 FILES is a collection, so it doesn't belong here
 */
 Variable:
-    '!'? ('ARGS_COMBINED_SIZE' | 'ARGS_GET_NAMES' | 'ARGS_NAMES' | 'ARGS_POST_NAMES' |
+    '!'? '&'? ('ARGS_COMBINED_SIZE' | 'ARGS_GET_NAMES' | 'ARGS_NAMES' | 'ARGS_POST_NAMES' |
     'AUTH_TYPE' | 'DURATION' | 'FILES_COMBINED_SIZE' | 'FILES_NAMES' | 'FILES' | 
     'FULL_REQUEST' | 'FULL_REQUEST_LENGTH' | 'FILES_SIZES' | 'FILES_TMPNAMES' | 
     'FILES_TMP_CONTENT' | 'HIGHEST_SEVERITY' | 'INBOUND_DATA_ERROR' | 'MATCHED_VAR_NAME' |
```

here:
```
$ cat issue_70.conf 
SecRule &REQUEST_BODY "@eq 1" \
     "id:941181,\
     phase:2,\
     block,\
     capture,\
     t:none"

$ poetry run secrules-parser -f issue_70.conf -c
Syntax OK: issue_70.conf
```

Without that:
```
$ poetry run secrules-parser -f issue_70.conf -c
Syntax invalid: Syntax error in line 1 col 10: Expected 'ARGS_GET' or 'ARGS_POST' or 'ARGS' or 'ENV' or 'GEO' or 'GLOBAL' or 'IP' or 'MATCHED_VARS_NAMES' or 'MATCHED_VARS' or 'MULTIPART_PART_HEADERS' or 'PERF_RULES' or 'REQUEST_COOKIES_NAMES' or 'REQUEST_COOKIES' or 'REQUEST_HEADERS_NAMES' or 'REQUEST_HEADERS' or 'RESPONSE_HEADERS_NAMES' or 'RESPONSE_HEADERS' or 'RULE' or 'SESSION' or 'TX' at position /home/airween/src/secrules_parsing/issue_70.conf:(1, 10) => 'SecRule &*REQUEST_BO'.
```

I don't see the point of mentioned line (and the whole collection count handling), because the `&` sign can preceded not just a collection but any kind of variable. I suggest we should drop that part, and add the `&` sign as an optional prefix before all variables.

What do you think?

**Comment by theseion** [comment]
Sounds good to me. You know best whether that's semantically correct.

**Comment by airween** [comment]
Solved via #71, so I close this issue.
